src/components/map.js

```diff
@@ -211,29 +227,96 @@ let Map = class Map extends React.Component {
       if (!this.props.mini) {
         this.map.on("mousemove", "sa2-fills", (e) => {
           if (e.features.length > 0) {
-            this.highlightFeature(e.features[0])
+            this.highlightFeature(e.features[0]);
           }
         });
-  
+
+        // Create clusters on load if selected for initial map loads to business layer
+        if (this.props.active.key === Constants.MAP_TYPE.BUSINESSES) {
+          this.drawBusinessClusters()
+        }
+
         // When the mouse leaves the sa2-fill layer, update the feature state of the
         // previously hovered feature.
-  
+
         this.map.on("mouseleave", "sa2-fills", this.clearFeatureHighlight);
-  
+
         // Handle clicks on map features
-        this.map.on("click", "sa2-fills", this.onMapClick)
+        this.map.on("click", "sa2-fills", this.onMapClick);
+
+        // Handle clicks on clusters
+        this.map.on("click", "clusters", this.zoomOnClick);
+
+        // Handle clicks on business points
+        this.map.on("click", "unclustered-point", this.pointOnClick);
+
+        // Add hover cursor for clusters
+        this.map.on("mouseenter", "clusters", this.setCursorPointer);
+        this.map.on("mouseleave", "clusters", this.setCursorNone);
+
+        // Add hover cursor for points
+        this.map.on("mouseenter", "unclustered-point", this.setCursorPointer);
+        this.map.on("mouseleave", "unclustered-point", this.setCursorNone);
+
         // Handle map clicks outside of map features
         this.map.on("click", this.onMapClick);
         if (this.props.selectedFeature) {
-          this.selectFeature(this.props.selectedFeature)
+          this.selectFeature(this.props.selectedFeature);
         }
-      }
-      else {
-        this.highlightComparisonFeatures(this.props.comparisonFeatures)
+      } else {
+        this.highlightComparisonFeatures(this.props.comparisonFeatures);
       }
     });
   }
 
+  pointOnClick = (e) => {
+    var coordinates = e.features[0].geometry.coordinates.slice();
+
+    const properties = e.features[0].properties;
+
+    // Ensure that if the map is zoomed out such that
+    // multiple copies of the feature are visible, the
+    // popup appears over the copy being pointed to.
+    while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
+      coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
+    }
+
+    new mapboxgl.Popup()
+      .setLngLat(coordinates)
+      .setHTML("<div class=\"popup-root\"><p class=\"popup-text-header\"> " + properties.name + "</p><br><p class=\"popup-text\">Type: " + properties.type + "</p></div>")
```

Comment:
If you get rid of the `<br>` here, there would be about the right amount of vertical space between the name and type.

Also, it'd be good to not include the text "Type: " as part of this.